### PR TITLE
Allows encryption_enabled to be turned off via environment variables:…

### PR DESCRIPTION
**Pull request by ireneybean**
… Issue #38

 - Allow encryption to be turned off by setting the ENCRYPTION_ENABLED environment variable to false, f, off, or no (case insensitive)

- Rspec tests

### CODE REVIEW

**Comment by ireneybean** [comment]
bah. one second.

File: lib/fake_idp/configuration.rb [hunk 1]
```diff
@@ -59,7 +59,7 @@ def default_algorithm
     end
 
     def default_encryption
-      ENV["ENCRYPTION_ENABLED"] || true
+      !%w[FALSE OFF F NO].include?(ENV["ENCRYPTION_ENABLED"].to_s.upcase) && true
```
Comment (samudary) [inline lib/fake_idp/configuration.rb:62]: @ireneybean First, thanks for contributing!

I think there may be an opportunity to simplify this and make encryption purely opt-in and instead default to false. Also, I think if this environment variable is set to `""` or `nil` (implicitly disabled) encryption would always be enabled which, as I think about this a bit more, might be unexpected for some users.

@cbortz What do you think?

Comment (ireneybean) [reply lib/fake_idp/configuration.rb:62]: Yep, turning encryption off with this patch would require specifically setting it to "false". It looked to me like y'all were trying have to have it default to true so I was trying to respect that, but allow it to be disabled via the environment variable. It doesn't make too much difference to me personally which is the default, as long as I can turn it off.  If having it default to off works for you, I'm good with that. I'll update this to work that way.

**Comment by ireneybean** [comment]
This will cause encryption to be off unless it's specifically turned on. Turning it on can be achieved via setting it to true in config block or by setting the environment variable to any non nil or empty value.  I feel a little weird about ENCRYPTION_ENABLED='false' still resulting in encryption being turned on, but I didn't want to overthink it either. I can add support for 'false' being false if you like. Let me know.

File: spec/configuration_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+require "spec_helper"
+require_relative "../lib/fake_idp/configuration"
+
+describe FakeIdp::Configuration do
+  it 'defaults encryption_enabled to false' do
+    expect(subject.encryption_enabled).to be_falsey
+  end
+
+  context 'when ENCRYPTION_ENABLED is set' do
+    before { ENV['ENCRYPTION_ENABLED'] = 'some_value' }
```
Comment (cbortz) [inline spec/configuration_spec.rb:10]: One minor tweak here that will limit the spec cleanup and prevent any uncaught side-effects would be to stub the `ENV` object, e.g.

```ruby
allow(ENV).to receive(:[]).with('ENCRYPTION_ENABLED').and_return('some_value')
```

This fully encapsulates our setup to a single stub that is self-contained to the parent RSpec block, and negates the need for the `after` callbacks. Super minor preference, no change actually required.

File: spec/configuration_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+require "spec_helper"
+require_relative "../lib/fake_idp/configuration"
+
+describe FakeIdp::Configuration do
```
Comment (cbortz) [inline spec/configuration_spec.rb:4]: 🙏 Thank you so much for adding this spec!

File: spec/fake_idp_spec.rb [hunk 4]
```diff
@@ -68,8 +68,8 @@
 
       it "sets encryption_enabled" do
         expect do
-          FakeIdp.configure { |config| config.encryption_enabled = false }
-        end.to change { FakeIdp.configuration.encryption_enabled }.from(true).to(false)
+          FakeIdp.configure { |config| config.encryption_enabled = true }
+        end.to change { FakeIdp.configuration.encryption_enabled }.from(false).to(true)
```
Comment (cbortz) [inline spec/fake_idp_spec.rb:72]: 🎉

File: README.md [hunk 5]
```diff
@@ -24,7 +24,7 @@ To run locally, you first need to set up the following environment variables:
 |NAME_ID|Name_id of the user you want to log in as - may be nil/blank|
 |SSO_UID|Unique id of the user you want to log in as - may be nil/blank|
 |USERNAME|Username of the user you want to log in as - may be nil/blank|
-|ENCRYPTION_ENABLED|Determines whether the generated assertion should be encrypted or not. Defaults to true|
+|ENCRYPTION_ENABLED| Set to 'true' to indicate that the generated assertion should be encrypted. Defaults to false|
```
Comment (cbortz) [inline README.md:27]: Super appreciative of the attention to detail and updating the README 👍

**Comment by cbortz** [comment]
@samudary I'll leave it to you to do what we need to do on our end (if anything) to get this merged in.

**Pull request merged**